tests/helpers.py

```diff
@@ -30,22 +31,31 @@ def run_app_with_settings(
             mocked and therefore the ``fmt`` argument ignored. You can use
             ``jinja2.Environment`` methods to create a ``Template`` or provide a template
             as a string.
+        hooks: An iterable of hooks to run while generating the documentation.
 
     Returns:
         Lower-cased stdout of the app.
     """
     if template:
         assert fmt == "markdown", "The `fmt` argument is ignored when `template` is used."
-        if isinstance(template, str):
-            template = Environment().from_string(template)
-        mocker.patch("settings_doc.main.get_template", return_value=template)
+
+        def get_template(env: Environment, _) -> Template:
```

Comment:
You couldn't have know because the coding standard I adhere to is not mentioned. But unused arguments are mentioned and explicitly deleted.
```suggestion
        def get_template(env: Environment, output_format: OutputFormat) -> Template:
            del output_format
```